packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3233 @@
+  ///
+  /// Used to avoid calling [MenuItem.onClose] unnecessarily.
+  void close() {
+    if (!isOpen) {
+      return;
+    }
+    isOpen = false;
+    item.onClose?.call();
+  }
+
+  // Used for testing to verify which item this is.
+  @override
+  String toStringShort({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    return item.toStringShort();
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[
+      ...children.map<DiagnosticsNode>((_MenuNode item) => item.toDiagnosticsNode()),
+    ];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<MenuItem>('item', item));
+    properties.add(DiagnosticsProperty<_MenuNode>('parent', parent, defaultValue: null));
+    properties.add(IntProperty('numChildren', children.length, defaultValue: null));
+  }
+}
+
+/// A controller that allows control of a [MenuBar].
+///
+/// Normally, it's not necessary to create a `MenuBarController` to use a
+/// [MenuBar], but if you need to be able to close any open menus with the
+/// [closeAll] method in response to an event, you can create one and pass it to
+/// the [MenuBar].
+///
+/// If the place you wish to close all the menus (for instance, when a control
+/// in one of the icons is selected), you can call [closeAll] on the controller
+/// to do so. If the control is a descendant of the [MenuBar], you don't need to
+/// create a `MenuBarController`, since the [MenuBar] will create one
+/// automatically. You can retrieve it with the [MenuBarController.of] method.
+abstract class MenuBarController with ChangeNotifier {
+  /// A factory that constructs a [MenuBarController] for use with a [MenuBar].
+  factory MenuBarController() => _MenuBarController();
+
+  // Private constructor to prevent this class from being instantiated by
+  // anything other than the factory constructor, and from being subclassed.
+  MenuBarController._();
+
+  /// Closes any menus that are currently open.
+  void closeAll();
+
+  /// Returns true if any menu in the menu bar is open.
+  bool get menuIsOpen;
+
+  /// Returns the active menu controller in the given context, and creates a
+  /// dependency relationship that will rebuild the context when the controller
+  /// changes.
+  static MenuBarController of(BuildContext context) {
+    final MenuBarController? found = context.dependOnInheritedWidgetOfExactType<_MenuBarControllerMarker>()?.controller;
+    if (found == null) {
+      throw FlutterError('A ${context.widget.runtimeType} requested a '
+          'MenuBarController, but was not a descendant of a MenuBar: $context');
+    }
+    return found;
+  }
+
+  /// A testing method used to provide access to a testing description of the
+  /// currently open menu for tests.
+  ///
+  /// Only meant to be called by tests.
+  @visibleForTesting
+  String? get testingCurrentItem;
```

Comment:
instead of "testing" call this "debug" for consistency with other places in the framework.

Also document and implement that this only does something in debug mode and in all other modes it will just return null or throw to discourage usage in non-test enfironments.